services/QuillLMS/spec/controllers/api/v1/concept_feedback_controller_spec.rb

```diff
@@ -27,6 +26,89 @@
           .reduce({}) { |agg, q| agg.update({q.uid => q.as_json}) }
       )
     end
+
+  end
+
+  describe "#translations" do
+    let(:locale) { Translatable::DEFAULT_LOCALE }
+    let!(:concept_feedback_untranslated) { create(:concept_feedback)}
+    let(:cache_key) {
+      "#{ConceptFeedback::ALL_CONCEPT_FEEDBACKS_KEY}_#{concept_feedback_untranslated.activity_type}_#{locale}"
+    }
+
+    context 'there is a translation available for the language' do
+      let!(:concept_feedback1) { create(:concept_feedback, :with_translated_text)}
+      let!(:concept_feedback2) { create(:concept_feedback, :with_translated_text)}
+
+      context 'there is no cache set' do
+        before do
+          $redis.del(cache_key)
+        end
+
+        it 'returns all the concept feedback translated fields for that language' do
+          get :translations, params: {
+              locale:,
+              activity_type: concept_feedback1.activity_type
+          }, as: :json
+          body = JSON.parse(response.body)
+          expect(body[concept_feedback1.uid]['description']).to eq(concept_feedback1.translation(locale:))
+          expect(body[concept_feedback2.uid]['description']).to eq(concept_feedback2.translation(locale:))
+          expect(body.keys).not_to include(concept_feedback_untranslated.uid)
+        end
+
+        it 'sets the redis cache' do
+          get :translations, params: {
+              locale:,
+              activity_type: concept_feedback1.activity_type
+          }, as: :json
+          expect($redis.get(cache_key)).to eq(response.body)
+        end
+      end
+
+      context 'the cache is set' do
+        let(:cache_value) { {foo: 'bar'}.to_json }
+
+        it 'returns the cache value' do
+          $redis.set(cache_key, cache_value)
```

Comment:
This is purely stylistic especially given that there's only the one `it` block here, but it might be worth standardizing whether these `$redis` calls are all going to be in `before` blocks (as they are above and below), or if we want them in the `it` blocks in these cases where there's only the one `it` block.